### DISCUSSION TITLE
Roadmap v1 [draft]

### DISCUSSION THREAD

**Discussion by StJudeWasHere** [wwwdevcc/projects#16] [Ideas]
In order to limit the scope for version 1 and have a clear path forward, this is a roadmap draft up for discussion.

This is what I believe users should be able to do:

- Anyone should be ablet to see existing projects.
- Users should be able to register and login.
- Registered user should be able to add new projects.
- Registered users should be able to edit their projects.
- Registered users should be able to remove their projects.

So the actual tasks could look something like this:

- User registration
- User login

- Projects listing
- Project add
- Project edit
- Project remove

For users, I would limit it to email and password. We could work on a complete user profile for next version.
For projects, I would keep it as we have it right now. Project name, description text, tech stack text, url, contact text (as it is now). For version 2 we could work on a notification system to let users get in touch.
On the front end I would keep the design and style simple and not worry too much about the looks, I think for version 1 we should focus on having something working, for future versions we could polish the design.

Then there're some other features that would be good to have, but not a priority:

- Password recovery
- User edit (change user name and password)

I think this limited scope would allow us to focus on a reduced set of features and ship fast. Once we reach version 1 adding new features should be much easier as we'd already have a working structure.

I've probably overlooked something, so feel free to comment and share your opinion.

**Comment by CydoEntis** [comment]
### **Some items I think you missed that are required for at least _Version 1_ of this app are
- Pagination
- Search
- Filtering

For filtering I think you should be able to filter by
- Created At
- Updated At
- Tech Stack
- Skill Level

At Project creation a user should also be able to include adding a skill level
- Beginner
- Intermediate
- Advanced
- Expert

Search can just be by project name to start 


### **These suggestions are for later iterations of the app**
- Expand search to other fields like contact.
- Link to a user page where you can see all their projects they created or are a part of.
- Project bookmarking or following (So you can keep track of projects you are interested in)
- Project Updates - As projects are being worked on a user can add something like "Patch Notes" so users can follow the development of a project.
-  Tech stack tabs, so instead of adding your whole tech stack, maybe predefined badges users can pick from. For example LAMP, MERN, MEAN, etc or badges of common languages/technologies like React, PHP, Postgres etc. with the option to enter a custom badge if yours isn't in the list.
